client/app/js/services.js

```diff
@@ -1277,6 +1283,108 @@ factory("Utils", ["$rootScope", "$http", "$q", "$location", "$filter", "$timeout
     }
   };
 }]).
+factory('mediaProcessor', ['Utils', function (Utils) {
+  return {
+    createWavBlob: function (modbuffer) {
+      var sampleRate = 25000;
+      const buffer = new ArrayBuffer(44 + modbuffer.length * 2);
+      const view = new DataView(buffer);
+
+      Utils.writeUTFBytes(view, 0, "RIFF");
+      view.setUint32(4, 44 + modbuffer.length * 2, true);
+      Utils.writeUTFBytes(view, 8, "WAVE");
+      Utils.writeUTFBytes(view, 12, "fmt ");
+      view.setUint32(16, 16, true);
+      view.setUint16(20, 1, true);
+      view.setUint16(22, 1, true);
+      view.setUint32(24, sampleRate, true);
+      view.setUint32(28, sampleRate, true);
+      view.setUint16(32, 2, true);
+      view.setUint16(34, 16, true);
+      Utils.writeUTFBytes(view, 36, "data");
+      view.setUint32(40, modbuffer.length * 2, true);
+
+      let index = 44;
+      for (let i = 0; i < modbuffer.length; i++) {
+        view.setInt16(index, modbuffer[i] * (0x7FFF), true);
+        index += 2;
+      }
+
+      return new Blob([view], { type: "audio/wav" });
+    },
+
+    applyGossipRemoval: function (buffer) {
+      const blockSize = 128; // Adjust the block size as needed (a power of 2 is recommended)
+      const numBlocks = Math.ceil(buffer.length / blockSize);
+
+      for (let blockIndex = 0; blockIndex < numBlocks; blockIndex++) {
+        const blockStart = blockIndex * blockSize;
+        const blockEnd = Math.min(blockStart + blockSize, buffer.length);
+
+        for (let i = blockStart; i < blockEnd; i++) {
+          if (Math.abs(buffer[i]) < 0.03) {
```

Comment:
@msmannan00 : which is the rationale behing this function and is there any documentation reference for this comparison? why exactly 0.03?